src/main/src/main/java/org/geoserver/config/GeoServerLoader.java

```diff
@@ -211,15 +211,18 @@ public void accept(LayerContents lc) {
                 catalog.add(l);
 
                 LOGGER.info("Loaded layer '" + l.getName() + "'");
-
+                List<StyleInfo> badStyles = new ArrayList<>();
                 for (StyleInfo style : l.getStyles()) {
                     if (null == style) {
                         LOGGER.log(
                                 Level.SEVERE,
                                 "Layer '" + l.getName() + "' references a missing style");
                     }
+                    badStyles.add(style);
```

Comment:
So this is collecting a list of nulls, which are all equals (actually all same) ... a boolean to track presence would do the same I guess. But I'm guessing it would not be same when removing them, removeAll should go an remove each one, otherwise if there are more than one, not all would be removed? :-D

Reply:
You're right they are all Null - I should have thought of that! As for what removeAll does with many Nulls? https://stackoverflow.com/questions/60417963/does-arraylist-removeallcollection-cares-about-duplicates seems to indicate removing a single Null should work - I'll simplfy